src/pilot/assign_role.py

```diff
@@ -1014,34 +1019,60 @@ def generate_osd_config(ip_mac_service_tag, drac_client):
 def get_drives(drac_client):
     spinners = []
     ssds = []
-    physical_disks = drac_client.list_physical_disks()
-    for physical_disk in physical_disks:
-        # Eliminate physical disks that in a state other than non-RAID
-        if physical_disk.raid_status != "non-RAID":
-            LOG.info("Skipping disk {id}, because it has a RAID status of "
-                     "{raid_status}".format(
-                         id=physical_disk.id,
-                         raid_status=physical_disk.raid_status))
-            continue
+    virtual_disks = drac_client.list_virtual_disks()
 
-        # Eliminate physical disks that have an error status
-        if physical_disk.status == 'error':
-            LOG.warning("Not using disk {id}, because it has a status of "
-                        "{status}".format(id=physical_disk.id,
-                                          status=physical_disk.status))
-            continue
+    raid0_disks = [vd for vd in virtual_disks if vd.raid_level != '1']
+
+    raid1_disks = []
+    for vd in virtual_disks:
+        if vd.raid_level == '1':
+            raid1_disks.extend(vd.physical_disks)
 
-        # Go ahead and use any physical drive that's not in an error state, but
-        # issue a warning if it's not in the ok or unknown state
-        if physical_disk.status != 'ok' and physical_disk.status != 'unknown':
-            LOG.warning("Using disk {id}, but it has a status of \""
-                        "{status}\"".format(id=physical_disk.id,
-                                            status=physical_disk.status))
+    # Getting all physical disks of raid_disks except RAID1 disks
+    physical_disks_of_raid = [pd for pd in drac_client.list_physical_disks()
+                              if pd.id not in raid1_disks]
 
-        if physical_disk.media_type == "hdd":
-            spinners.append(physical_disk)
+    for virtual_disk in raid0_disks:
+        phy_disk = [pd for pd in physical_disks_of_raid
+                    if virtual_disk.physical_disks[0] == pd.id][0]
+
+        if phy_disk.media_type == 'hdd':
+            spinners.append(virtual_disk)
         else:
-            ssds.append(physical_disk)
+            ssds.append(virtual_disk)
+
+        physical_disks_of_raid.remove(phy_disk)
```

Comment:
I think there is a bug here.  When we exit this loop, physical_disks_of_raid should contain only physical disks that are not in any RAID for the block of code below to work correctly.  The line above only removes 1 physical disk that backs the current virtual disk from physical_disks_of_raid though, when all of the physical disks that back the current virtual disk should be removed.  I'm thinking that lines 1036 and 1037 should be simplified to just make a list of all physical disks that back the RAID.  I'm thinking that physical_disks_of_raid should be a dictionary that maps pd.id to pd.  That would make lines 1036 & 1037 something like:
phy_disks = [physical_disks_of_raid[pd_id] for pd_id in virtual_disk.physical_disks]
Then we could use phy_disks here to delete all of the disks that back the virtual disk from physical_disks_of_raid.